src/nupic/regions/TMRegion.cpp

```diff
@@ -79,20 +79,57 @@ TMRegion::TMRegion(const ValueMap &params, Region *region)
 
   args_.iter = 0;
   args_.sequencePos = 0;
-  args_.outputWidth = (args_.orColumnOutputs)?args_.numberOfCols 
+  args_.outputWidth = (args_.orColumnOutputs)?args_.numberOfCols
```

Comment:
just a question: I assume the "orColumnsOutput" does a logical OR (union) over all cells in a column(?) 
@dkeeney is this possible in the algorithm too, or only in Region? (would be useful in both) 
@ctrl-z-9000-times how is topology taken care of there? a sequential (cellsPerColumn = C) cells does not correspond to a topological column (as you've said, only TP has Columns, but not TM,CP). This is a problem, as Anomaly does need columns,  not cells.

Reply:
> How is topology taken care of there?

IIRC: The TM flattens it's input mini-columns into a single dimension.  The TM has no concept of topology.

> (as you've said, only TP has Columns, but not TM,CP)

No.  This should read:
Only TM has Columns, but not SP, TP/CP

BTW: TP == CP. Same concept and same location in brain, different algorithms.

Reply:
> It is not an argument passed into the algorithm but it is an 'algorithm sort of thing'. The code snippet above is executed for all 4 variations of the TM algorithm. It could be that it was intended to only be used with the BacktrackingTM. 

anomaly needs columns with cells to work correctly, so I'd like to move this down to algo, where it should be. It works for TP, BackTM; TM needs bigger rework to do that. 

> The configured topology in the TMRegion is maintained for the Column portion. An extra lowest level dimension is added for the outputs that return cells. i.e. dimensions [6, 4] becomes [100, 6, 4] if 100 cells per column.

:+1:  thanks, that's perfect